### DISCUSSION TITLE
Any plans to upgrade zustand to 4?

### DISCUSSION THREAD

**Discussion by kayhanb** [udecode/zustand-x#53] [General]
zustand has now upgraded to 4.3.1 and zustood is currently using ^3.6.4. When I upgraded zustand to 4.3.1 in my local I was warned that some approaches will be deprecated. Does this excellent library have any plans to adapt to it?

Warnings:
<img width="1253" alt="screenshot" src="https://user-images.githubusercontent.com/29417376/212033683-6fed9803-86f3-46c6-ac19-69b97702661f.png">

**Comment by zbeyens** [comment]
Yes

**Comment by polypixeldev** [comment]
What's the progress on this? Is any help needed?

**Reply by andreifloricel** [reply]
Hi @zbeyens , can we help in any way for the update to zustand 4.x?

**Reply by zbeyens** [reply]
Sure, please go on, I'll release it asap

**Reply by ZipBrandon** [reply]
Has there been any traction on this yet or does it still need to be initiated?

**Comment by enchorb** [comment]
Any ETA on this?

**Reply by zbeyens** [reply]
I got a blocker on my first try, any help is welcome.

**Reply by ZipBrandon** [reply]
Sure! Is there a branch or more info on the blocker?

**Reply by zbeyens** [reply]
Super! There is no branch but we basically need to migrate the types. `UseStore` is the starting point – not exported by zustand anymore. https://docs.pmnd.rs/zustand/migrations/migrating-to-v4

**Reply by zbeyens** [reply]
https://github.com/InfiniteXyy/zoov is using zustand 4 and has similar types. It may help.

**Comment by enchorb** [comment]
PR: https://github.com/udecode/zustood/pull/57
